vendor_blueprints/export_card_to_file.py

```diff
@@ -0,0 +1,181 @@
+        card_ids += cards
+
+    return card_ids
+
+
+def get_card_data(card_id, access_token):
+    """
+    Get metadata and property information of a single card
+
+    card_id (str): The unique ID of the card
+    access_token: client access token. Use the get_access_token() function to retrieve the token.
+
+    Returns:
+    card_response -> dict with the metadata and details of the card.
+    """
+    card_info_api = f"https://{DOMO_INSTANCE}.domo.com/api/content/v1/cards?urns={card_id}&parts=metadata,properties&includeFiltered=true"
+    card_headers = {
+            'Content-Type' : 'application/json',
+            'x-domo-authentication': access_token
+        }
+    card_response = requests.get(url=card_info_api, headers=card_headers)
+    return card_response.json()
+
+
+def get_card_id_from_name(name, client_id, secret_key, access_token):
+    """
+    Retrieve the particular Card Id of a card given the name
+    """
+    cards = get_all_card_ids()
+    for card_id in cards:
+        card = get_card_data(card_id, access_token)
+        card_name = card[0]['title']
+        # check if name matches with card name (ignore all caps)
+        if re.search(name, card_name, re.IGNORECASE):
+            return card_id
+
+
+def export_graph_to_file(card_id, filename, file_type, access_token):
+    """
+    Exports a file to one of the given file types: csv, ppt, excel
+    """
+    export_api = f"https://{DOMO_INSTANCE}.domo.com/api/content/v1/cards/{card_id}/export"
+
+    card_headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'accept': 'application/json, text/plain, */*',
+            'x-domo-authentication': access_token
+    }
+    
+    # make a dictionary to map user file_type with requested mimetype
+    filetype_map = {
+        "csv":"text/csv", 
+        "excel":"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        "ppt": "application/vnd.ms-powerpoint"
+    }
+    
+    body = {
+        "queryOverrides":{
+            "filters":[],
+            "dataControlContext":{
+                "filterGroupIds":[]
+            }
+        },
+        "watermark":'true',
+        "mobile":'false',
+        "showAnnotations":'true',
+        "type":"file",
+        "fileName": f"{filename}",
+        "accept": filetype_map[file_type]
+    }
+    # convert body to domo encoded payload
+    encoded_body = urllib.parse.quote(f"{body}")
+    encoded_body = encoded_body.replace("%27", "%22") # changing " to '
+    payload = f"request={encoded_body}"
+
+    export_response = requests.post(url=export_api, data=payload,
+        headers=card_headers, stream=True)
+    if export_response.status_code == 200:
+        with open(filename, 'wb') as fd:
+            # iterate through the blob 1MB at a time
+            for chunk in export_response.iter_content(1024*1024):
+                fd.write(chunk)
+        print(f"{file_type} file:{filename} saved successfully!")
```

Comment:
Need to use Shipyard utils here for file destination. User should be able to provide a destination-file-name AND a destination-folder-name. If a folder name is provided, the folder should be created in the current working directory and file should be created there. See other blueprints for examples.